For both parts of the assignment

parthgoel17 opened the pull request:
The single inverted pendulum files are inside its folder and there is a separate package for the double inverted pendulum.

parthgoel17 committed:
Add files via upload

parthgoel17 committed:
Delete double_pendulum.py

parthgoel17 committed:
Rename pendulum_balance.py to single_inverted/single_inverted/pendulum_balance.py

changed folder

parthgoel17 committed:
Rename pendulum_swing.py to single_inverted/single_inverted/pendulum_swing.py

changed folder

parthgoel17 committed:
Rename pendulum_interface.py to single_inverted/single_inverted/pendulum_interface.py

parthgoel17 committed:
added double pendulum package

parthgoel17 committed:
updated forked repository

parthgoel17 committed:
Update torque saturation

parthgoel17 committed:
updated toque saturation

parthgoel17 committed:
Merge branch 'main' into main

deepboliya commented:
You should run ros2 nodes using "ros2 run..." command rather than "python3". And then add it to the launch file as well. The balancing is working. but pendulum_swing.py and pendulum_balance.py are exactly the same file. So how are you achieving swing up. Your assumption of dt=1/500 is wrong and also rather than assuming a constant dt you should calculate your loop time in real time, balancing still works tho because it acts like a another constant in multiplication with kp ki kd

parthgoel17 committed:
Update pendulum_balance.py

parthgoel17 committed:
Update pendulum_swing.py

parthgoel17 committed:
Update single_inverted_pendulum.launch.py

parthgoel17 committed:
Update single_inverted_pendulum.launch.py

parthgoel17 commented:
I wasn't aware that i can it add to the launch file as well. and i run my nodes by the command ros2 run <package-name> <executable-name>. I have updated the 'dt' term. As for the swing up, I didnt understand what the difference between their motions are, I thought it was just the change in initial state that matters. Can you explain what the difference both controllers are trying to achieve?

deepboliya commented:
> I wasn't aware that i can it add to the launch file as well. and i run my nodes by the command ros2 run . I have updated the 'dt' term. As for the swing up, I didnt understand what the difference between their motions are, I thought it was just the change in initial state that matters. Can you explain what the difference both controllers are trying to achieve?

The fact that we don't have enough torque changes everything, if you use PID, then the torque just remains clipped at 5. So you need to devise a method to increase energy of the pendulum such that it is able to reach the top most point and then switch to PID controller

deepboliya commented:
Also add a diagram for your assumption of theta_0 and theta_1

parthgoel17 committed:
Update pendulum_swing.py

updated for swing up initially followed by PID

parthgoel17 committed:
Add files via upload

parthgoel17 committed:
Delete balance_initial_conditions.HEIC

parthgoel17 committed:
Delete swing_initial_conditions.heic

parthgoel17 committed:
Add files via upload

parthgoel17 committed:
Rename balance_initial_conditions.jpg to illustrations/balance_initial_conditions.jpg

parthgoel17 committed:
Rename swing_initial_conditions.jpg to illustrations/swing_initial_conditions.jpg

parthgoel17 committed:
Update README.md

parthgoel17 committed:
Update pendulum_swing.py

used a flag to tell if swing up completed

parthgoel17 commented:
I have added a flag which is false until swing up is completed and during that time, i use a torque to increase the energy of the system. as soon as the swing up is completed, it switches to the PD controller to balance it. Added diagrams for initial conditions in both scenarios as well.

parthgoel17 committed:
Update pendulum_swing.py

parthgoel17 committed:
Update setup.py

parthgoel17 commented:
Please run the pendulum_swing.py node for the swing up function. and sorry, I forgot to commit my setup.py file ;/. I'll add the diagram for the double inverted pendulum as well.

parthgoel17 committed:
Add files via upload

parthgoel17 committed:
Delete racing-2024-06-09_19.58.53
